pkg/server/message.go

```diff
@@ -78,13 +81,31 @@ func (t *TranslateServiceServer) CreateMessages(
 		return nil, status.Errorf(codes.AlreadyExists, "messages already exist for language: '%s'", params.messages.Language)
 	}
 
-	// If the messages are not the original messages, translate them.
+	// Translate messages when messages are not original and original language is known.
 	if !params.messages.Original {
-		var err error
-
-		params.messages, err = t.translator.Translate(ctx, params.messages)
+		// Retrieve language from original messages.
+		var originalLanguage *language.Tag
+		// TODO: to improve performance should be replaced with CheckMessagesExist db function.
+		msgs, err := t.repo.LoadMessages(ctx, params.serviceID, common.LoadMessagesOpts{})
 		if err != nil {
-			return nil, status.Errorf(codes.Unknown, err.Error()) // TODO: For now we don't know the cause of the error.
+			return nil, status.Errorf(codes.Internal, "")
+		}
+
+		for _, v := range msgs {
+			if v.Original {
+				originalLanguage = &v.Language
+			}
+		}
```

Comment:
```suggestion
		for _, v := range msgs {
			if v.Original {
				originalLanguage = &v.Language
        break
			}
		}
```
Original is only one (hopefully), so no need to continue iterating